flytepropeller/pkg/controller/nodes/array/handler.go

```diff
@@ -500,13 +500,18 @@ func (a *arrayNodeHandler) Handle(ctx context.Context, nCtx interfaces.NodeExecu
 					}{nil, err}
 					continue
 				}
-
 				// checkpoint paths are not computed here because this function is only called when writing
 				// existing cached outputs. if this functionality changes this will need to be revisited.
 				outputPaths := ioutils.NewCheckpointRemoteFilePaths(ctx, nCtx.DataStore(), subOutputDir, ioutils.NewRawOutputPaths(ctx, subDataDir), "")
-				reader := ioutils.NewRemoteFileOutputReader(ctx, nCtx.DataStore(), outputPaths, 0)
-
-				gatherOutputsRequest.reader = &reader
+				reader, err := ioutils.NewRemoteFileOutputReader(ctx, nCtx.DataStore(), outputPaths, 0)
+				if err != nil {
+					gatherOutputsRequest.responseChannel <- struct {
```

Comment:
I have to admit I don't understand this change, could you please explain?

Reply:
`NewRemoteFileOutputReader` has to return an error now, if the error path is mis-structured, because it uses some of the parsing methods for `DataReference` that can potentially return an error.

The original constructor was not returning an error. So with the new structure, the error case needs to be handled. I don't know what the `gatherOutputsRequest.responseChannel <- struct` thingie does, but if you go up a little in the file, you'll see multiple examples of doing exactly that (repeated) when there is an unexpected error. So I just followed the convention already used in the file.

Reply:
Addressed.